src/neuromaps_nhp/graph.py

```diff
@@ -0,0 +1,394 @@
+                    rad = 0.1 + (j - (num_edges - 1) / 2) * 0.15
+                    connectionstyle = f"arc3,rad={rad}"
+
+                nx.draw_networkx_edges(
+                    self.graph,
+                    pos,
+                    edgelist=[(source, target)],
+                    edge_color=[edge_colors[edge_idx]],
+                    arrows=True,
+                    arrowsize=15,
+                    arrowstyle="->",
+                    alpha=0.7,
+                    connectionstyle=connectionstyle,
+                    width=2,
+                    min_source_margin=30,
+                    min_target_margin=30,
+                )
+
+        # ===== LEGENDS =====
+
+        # Create edge legend
+        edge_legend_elements = []
+        for density, color in density_colors_map.items():
+            edge_legend_elements.append(
+                Line2D([0], [0], color=color, lw=3, label=f"{density} density")
+            )
+
+        if edge_legend_elements:
+            plt.legend()
+
+        # Enhanced title
+        total_transforms = len(combined_edges)
+        plt.title(
+            f"Brain Template Surface Transform Graph\n"
+            f"Spaces: {self.graph.number_of_nodes()}, "
+            f"Transforms: {total_transforms}",
+            fontsize=14,
+            fontweight="bold",
+            pad=20,
+        )
+
+        plt.axis("off")
+        plt.tight_layout()
+        plt.savefig(
+            save_path, dpi=300, bbox_inches="tight", facecolor="white", edgecolor="none"
+        )
+        print(f"Graph visualization saved to: {save_path}")
+
+    def _generate_color_palette(
+        self, items: List[str], colormap: str = "tab10"
+    ) -> Dict[str, str]:
+        """Generate a dynamic color palette for a list of items."""
+        if not items:
+            return {}
+
+        # Use matplotlib colormaps for consistent colors
+        if len(items) <= 10:
+            colors = cm.tab10(np.linspace(0, 1, len(items)))
+        elif len(items) <= 20:
+            colors = cm.tab20(np.linspace(0, 1, len(items)))
+        else:
+            # For many items, use a continuous colormap
+            colors = cm.viridis(np.linspace(0, 1, len(items)))
+
+        # Convert to hex colors for consistency
+        hex_colors = [mcolors.to_hex(color) for color in colors]
+
+        return dict(zip(sorted(items), hex_colors))
+
+    def find_path(self, source: str, target: str) -> List[str]:
+        """Find shortest path between two spaces."""
+        try:
+            path = nx.shortest_path(self.graph, source, target)
+            return list(path)
+        except nx.NetworkXNoPath:
+            return []
+
+    def has_path(self, source: str, target: str) -> bool:
+        """Check if path exists between two spaces."""
+        result = nx.has_path(self.graph, source, target)
+        return bool(result)
+
+
+def graph_yaml(yaml_path: Path) -> SurfaceGraph:
```

Comment:
Just a note here (cc: @tamsinrogers) - will want to include methods for adding the volumetric stuff in here too. Once we have the `volume_graph.yaml` also included, I can combine them and either hard code that somewhere as a variable or just have a yaml file to load.